apps/examples/testcase/le_tc/kernel/itc_pthread.c

```diff
@@ -0,0 +1,505 @@
+		ret_chk = pthread_setspecific(g_key, &exec_index);
+		if (ret_chk != OK) {
+			printf("itc_pthread_key_create_set_getspecific_p_multitime: TC FAIL\n");
+			return NULL;
+		}
+
+		get_spec = pthread_getspecific(g_key);
+		if (*get_spec != exec_index) {
+			printf("itc_pthread_key_create_set_getspecific_p_multitime: TC FAIL\n");
+			return NULL;
+		}
+	}
+	sleep(WAIT_TIME_1);
+	g_flag = true;
+	return NULL;
+}
+
+/**
+* @fn                   :task_thread
+* @brief                :utility function for itc_pthread_pthread_kill_n_rekill
+* @return               :void*
+*/
+static void *task_thread(void *param)
+{
+	sleep(WAIT_TIME_8);
+	pthread_setname_np(0, "task_thread");
+	pthread_exit(RETURN_PTHREAD_JOIN);
+	return NULL;
+}
+
+/****************************************************************************
+  * Private Functions
+ ****************************************************************************/
+
+/**
+* @fn                   :itc_pthread_pthread_barrier_destroy_n_redestory
+* @brief                :The pthread_barrier_init() function allocates any resources required to use
+*                        the barrier referenced by 'barrier' and initialized the barrier with the
+*                        attributes referenced by attr.
+*                        barrier_wait synchronizse participating threads at the barrier referenced by 'barrier'.
+* @Scenario             :pthread_barrier_destroy re-destroying the object as already being destroyed
+* API's covered         :pthread_barrier_init, pthread_barrier_wait , pthread_barrier_destroy
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_pthread_pthread_barrier_destroy_n_redestory(void)
+{
+	int ret;
+	pthread_barrier_t pthread_barrier;
+	int cnt = 1;
+	
+	ret = pthread_barrier_init(&pthread_barrier, 0, cnt);
+	TC_ASSERT_EQ("pthread_barrier_init", ret, OK);
+
+	ret = pthread_barrier_destroy(&pthread_barrier);
+	TC_ASSERT_EQ("pthread_barrier_destroy", ret, OK);
+
+	ret = pthread_barrier_destroy(&pthread_barrier);
+	TC_ASSERT_EQ("pthread_barrier_destroy", ret, EINVAL);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_pthread_pthread_kill_n_rekill
+* @brief                :send a signal to a thread
+* @Scenario             :The pthread_kill() call again when it's already being called for same thread
+* API's covered         :pthread_kill, pthread_join
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_pthread_pthread_kill_n_rekill(void)
+{
+	int ret_chk;
+	bool bpthreadcallback = false;
+	pthread_t thread;
+
+	ret_chk = pthread_create(&thread, NULL, task_thread, NULL);
+	
+	TC_ASSERT_EQ("pthread_create", ret_chk, OK);
+	sleep(WAIT_TIME_2);
+
+	ret_chk = pthread_kill(thread, SIGUSR1);
+	sleep(WAIT_TIME_1);
+	TC_ASSERT_EQ_CLEANUP("pthread_kill", ret_chk, OK, pthread_detach(thread));
+	TC_ASSERT_EQ_CLEANUP("pthread_kill", bpthreadcallback, true, pthread_detach(thread));
+
+	ret_chk = pthread_kill(thread, SIGUSR1);
```

Comment:
```pthread_kill(thread, SIGUSR1);``` is sending ```SIGUSR1``` signal to ```thread```. Does it return ESRCH? Is it right?

Reply:
pthread_kill checks wether the thread ID exists or not if not then it returns ESRCH,as the thread has been killed in the previous kill instruction.

Reply:
No. I got a message from chat. Please check it again.
```
[장성한 / Sung-Han Chang] - 20:02 
And pthread_kill is not a function to kill thread

[장성한 / Sung-Han Chang] - 20:02 
Please read posix description
```